### PR TITLE
Bug fix 3.6/reset shadowrows block on return

### PR DESCRIPTION
**Pull request by markuspf**
Backport of #10834

### CODE REVIEW

**Comment by markuspf** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/7923/

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7935/

**Comment by mchacki** [comment]
Relevant test blue, the Mac community test is due to hardware and this PR does not contain anything that is community/enterprise specific

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr-mac-onlycatch/5828/

File: CHANGELOG [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+* Fix bug affecting spliced subqueries when memory blocks are reused.
+
 * Fixed internal issue #656: While executing large amount of concurrent
   insert/removes iresearch seems to leak open file handles. This results in
   error 32 in cleanup (observed only on Windows as Linux doesn`t have file
   sharing locks).
 
 * Updated arangosync to 0.7.2.
 
+* Hide swagger api calls in list by default
```
Comment (jsteemann) [inline CHANGELOG]: this CHANGELOG entry does not belong into 3.6.

**Comment by mchacki** [comment]
@KVS85 the linked Jenkins failed because of the system being broken.
In this PR there is no difference on Mac for Community vs. Enterprise that would not be covered by the enterprise branch.
I would consider this as good.

**Pull request merged**